quill-core/src/test/scala/io/getquill/sources/ExtractEntityAndInsertActionSpec.scala

```diff
@@ -0,0 +1,14 @@
+package io.getquill.sources
+
+import io.getquill.Spec
+import io.getquill.ast._
+
+class ExtractEntityAndInsertActionSpec extends Spec {
+  "Extract should work" in {
+    val ast = Map(Insert(Entity("TestEntity", Some("test"), List(PropertyAlias("i", "'i"), PropertyAlias("o", "'o")), Some("i"))), Ident("o"), Ident("o"))
```

Comment:
I know, I used the raw AST version so I could visualize how the transformation would be done.